tests/subsys/zigbee/cmake/extensions.cmake

```diff
@@ -0,0 +1,17 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+function(zigbee_get_zboss_include_path include_path)
```

Comment:
i'm not really fond of this solution.

It starts to build knowledge of Kconfig paths inside tests, something that really shouldn't be needed here.

I notice a couple of details, not directly related in this PR, but impacting this code.
The two symbols `ZIGBEE_LIBRARY_PRODUCTION` and `ZIGBEE_LIBRARY_DEVELOPMENT` are not depending on the `ZIGBEE` symbol, which looks a bit weird.
Why being able to specify lib type for something not used.

I know you don't need to link to the actual Zigbee lib for those tests, but be maybe that should be handled differently. If that is cleaned up, then you could have `ZIGBEE=y` which will give you the `zboss` interface lib, which has the correct headers to use (production/development).

As you just need the headers, you can then then do:
```
target_include_directories(app PRIVATE
  ${NRF_DIR}/subsys/zigbee/osif
  $<TARGET_PROPERTY:zboss,INTERFACE_INCLUDE_DIRECTORIES>
)
```

and there would be no need for:
- Any knowledge of `ZIGBEE_LIBRARY_PRODUCTION` vs `ZIGBEE_LIBRARY_DEVELOPMENT`
- No need for any `zigbee_get_zboss_include_path` function.

This might need some adjustment to how you handle the `APP_LINK_WITH_ZIGBEE` symbol.
Which, by the way, is prompt-less. the `APP_LINK_WITH_<name>` should usually have a prompt, so that end-user can decide if linking to app should be performed or not.
(user may want to link to another lib than app).

Reply:
PR has been rewritten, cmake extension is removed and the path to zboss headers has been aligned with the production ZBOSS in nrfxlib. I've added a separate commit to update manifest with nrfxlib fix that makes zboss KConfig symbols, including `ZIGBEE_LIBRARY_PRODUCTION `and `ZIGBEE_LIBRARY_DEVELOPMENT`, dependent on `ZIGBEE` symbol.